openapi/src/main/java/io/micronaut/openapi/visitor/ConfigProperty.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.openapi.visitor;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Supported configuration properties.
+ *
+ * @since 4.10.0
+ */
+public interface ConfigProperty {
```

Comment:
I would rename this to something less generic like `OpenApiConfigProperty` or something

Reply:
@graemerocher fixed